packages/map-toolkit/src/map-mode/store.ts

```diff
@@ -0,0 +1,485 @@
+   * ```
+   */
+  requestModeChange = (desiredMode: string, requestOwner: string): void => {
+    if (!desiredMode?.trim()) {
+      throw new Error('requestModeChange requires non-empty desiredMode');
+    }
+    if (!requestOwner?.trim()) {
+      throw new Error('requestModeChange requires non-empty requestOwner');
+    }
+
+    this.bus.emit(MapModeEvents.changeRequest, {
+      desiredMode: desiredMode.trim(),
+      owner: requestOwner.trim(),
+      instanceId: this.instanceId,
+    });
+  };
+
+  /**
+   * Notify all subscribers of state change
+   */
+  private notify(): void {
+    for (const listener of this.listeners) {
+      listener();
+    }
+  }
+
+  /**
+   * Setup event listeners for bus events
+   *
+   * Note: Event listeners remain active even after early returns in handlers.
+   * This is by design - cleanup happens in destroy() which is called automatically
+   * by MapIdProvider on unmount. Consumers don't need to manually manage cleanup.
+   */
+  private setupEventListeners(): void {
+    // Listen for mode change requests
+    const unsubRequest = this.bus.on(MapModeEvents.changeRequest, (event) => {
+      const { desiredMode, owner: requestOwner, instanceId } = event.payload;
+
+      // Filter: only handle if targeted at this instance
+      if (instanceId !== this.instanceId || desiredMode === this.mode) {
+        return;
+      }
+
+      this.handleModeChangeRequest(desiredMode, requestOwner);
+    });
+    this.unsubscribers.push(unsubRequest);
+
+    // Listen for authorization decisions
+    const unsubDecision = this.bus.on(MapModeEvents.changeDecision, (event) => {
+      const { instanceId, approved, authId, owner } = event.payload;
+
+      // Filter: only handle if targeted at this instance
+      if (instanceId !== this.instanceId) {
+        return;
+      }
+
+      this.handleAuthorizationDecision({ approved, authId, owner });
+    });
+    this.unsubscribers.push(unsubDecision);
+
+    // Listen for mode changes to handle pending requests
+    const unsubChanged = this.bus.on(MapModeEvents.changed, (event) => {
+      const { currentMode, previousMode, instanceId } = event.payload;
+
+      // Filter: only handle if targeted at this instance
+      if (instanceId !== this.instanceId) {
+        return;
+      }
+
+      // When mode owner changes to default mode, handle pending requests
+      if (currentMode === this.defaultMode && this.pendingRequests.size > 0) {
+        this.handlePendingRequestsOnDefaultMode(previousMode);
+      }
+    });
+    this.unsubscribers.push(unsubChanged);
+  }
+
+  /**
+   * Determine if a mode change request should be auto-accepted without authorization
+   */
+  private shouldAutoAcceptRequest(
+    desiredMode: string,
+    requestOwner: string,
+    currentModeOwner: string | undefined,
+    desiredModeOwner: string | undefined,
+  ): boolean {
+    // Owner returning to default mode
```

Comment:
NIT (I bet other people would disagree with me): could be one return statement for the whole thing...